src/lib/CustomRulePathManager.ts

```diff
@@ -119,3 +128,17 @@ export class CustomRulePathManager {
     return json;
   }
 }
+
+// Exported only for test visibility
+  // eslint-disable-next-line @typescript-eslint/no-use-before-define
+export class FileOperations {
```

Comment:
Nitpick: I'm not in love with this name, but I have no idea what a better name would be. My personal preference is something ending in 'handler' or 'manager'.

Reply:
Will rename it to FileOperationsHandler